Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Beckn/Common.hs

```diff
@@ -478,8 +478,9 @@ validateRideAssignedReq ::
   RideAssignedReq ->
   m ValidatedRideAssignedReq
 validateRideAssignedReq RideAssignedReq {..} = do
+  let BookingDetails {..} = bookingDetails
   booking <- QRB.findByTransactionId transactionId >>= fromMaybeM (BookingDoesNotExist $ "transactionId:-" <> transactionId)
-  unless (isAssignable booking) $ throwError (BookingInvalidStatus $ show booking.status)
+  unless (isAssignable booking || isInitiatedByCronJob) $ throwError (BookingInvalidStatus $ show booking.status)
```

Comment:
Not required for `RIDE_ASSIGNED`.

Reply:
can you please explain why it is not required in the RIDE_ASSIGNED validation ?

Reply:
CronJob only tries to bring ride and booking to terminal state, this events don't lead to terminal state.